modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Fields.tsx

```diff
@@ -433,8 +491,15 @@ const Fields = ({fdsView, fdsViewsURL}: FDSViewSectionInterface) => {
 			contentComponent: ({closeModal}: {closeModal: Function}) => (
 				<AddFDSFieldsModalContent
 					closeModal={closeModal}
+					fdsFields={fdsFields || []}
 					fdsView={fdsView}
-					onSave={() => getFDSFields()}
+					namespace={namespace}
+					onSave={() => {
+						setFDSFields(null);
+
+						getFDSFields();
```

Comment:
This seems odd. Just thinking through how I read this "When the modal is saved, fields gets set to null and then fetch fields"

I would think that onSave might return the new set of fields from the response of the save action

Reply:
The `null` part is a bit of my experiment in this file, to see how the pattern feels. Instead of having the separate `loading` state variable, I used value of `null` to reset to loading indicator in UI. I'm still not sure if it's better or worse.

Separate from that, I completely agree, we can do better than reload after save, we can return newly persisted info. This is as also noted in https://github.com/liferay-frontend/liferay-portal/pull/3288#discussion_r1180502026 and https://github.com/liferay-frontend/liferay-portal/pull/3288#discussion_r1181537946

Reply:
Ah gotcha. Yeah it's not a major issue, just was trying to wrap my head around it while reading it.

We may consider trying to use the [Suspense API](https://react.dev/reference/react/Suspense) a bit too for loading indicators. I haven't touched it much though, so idk if we need any specific architecture to utilize it.